module/cloud-services/files/helper.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * name : files/helper.js
+ * author : prajwal
+ * created-date : 25-Apr-2024
+ * Description : All files related helper functionality.Including uploading file
+ * to cloud service.
+ */
+
+// Dependencies
+
+let filesHelpers = require(MODULES_BASE_PATH + '/files/helper')
+const cloudStorage = process.env.CLOUD_STORAGE_PROVIDER
+const bucketName = process.env.CLOUD_STORAGE_BUCKETNAME
+const bucktType = process.env.CLOUD_STORAGE_BUCKET_TYPE
+const fs = require('fs')
+
+/**
+ * FilesHelper
+ * @class
+ */
+
+module.exports = class FilesHelper {
+	/**
+	 * Get all signed urls.
+	 * @method
+	 * @name preSignedUrls
+	 * @param {Array} payloadData       - payload for files data.
+	 * @param {String} referenceType    - reference type
+	 * @param {String} userId           - Logged in user id.
+	 * @param {String} templateId       - certificateTemplateId.
+	 * @param {Boolean} serviceUpload     - serive Upload  {true/false}
+	 * @returns {Array}                 - consists of all signed urls files.
+	 */
+
+	static preSignedUrls(payloadData, userId = '') {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let payloadIds = Object.keys(payloadData)
+
+				let result = {
+					[payloadIds[0]]: {},
+				}
+
+				for (let pointerToPayload = 0; pointerToPayload < payloadIds.length; pointerToPayload++) {
+					let payloadId = payloadIds[pointerToPayload]
+					// Generate unique folderPath to all the file names in payloadData
+					let folderPath = 'project/' + payloadId + '/' + userId + '/' + UTILS.generateUniqueId() + '/'
+					// Call preSignedUrls helper file to get the signedUrl
+					let imagePayload = await filesHelpers.preSignedUrls(
+						payloadData[payloadId].files,
+						bucketName,
+						cloudStorage,
+						folderPath,
+						typeof process.env.PRESIGNED_URL_EXPIRY_IN_SECONDS === 'string'
```

Comment:
@Prajwal17Tunerlabs do we need this check? we can convert to int even if it is a number. avoid the check always convert to int. make sure to test giving as string and link is expiring after the expiry time

Reply:
removed the check